MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -1583,6 +1595,9 @@
 	if ( [_mtlDevice respondsToSelector: @selector(maxThreadgroupMemoryLength)] ) {
 		_properties.limits.maxComputeSharedMemorySize = (uint32_t)_mtlDevice.maxThreadgroupMemoryLength;
 	} else {
+#if MVK_TVOS
+		_properties.limits.maxComputeSharedMemorySize = (32 * KIBI);
```

Comment:
This might not be right either.

Reply:
Only Apple family 4 (A11) and up support 32 kiB threadgroup-shared memory. Family 3 (A10, `MTLFeatureSet_tvOS_GPUFamily2*`) and below only support 16 kiB, and Family 2 (A8/A9, `MTLFeatureSet_tvOS_GPUFamily1*`)'s drivers actually reserve 32 bytes for themselves. (This latter detail is explained in a footnote in the Metal 2.1 feature set tables I sent you.)

Reply:
Thanks!